packages/frontend/src/pages/Dashboard.tsx

```diff
@@ -201,50 +202,46 @@ const Dashboard: FC = () => {
         [dashboardTiles, isEditMode],
     );
 
-    const { tiles: savedTiles } = dashboard || {};
     useEffect(() => {
-        if (savedTiles) {
-            clearIsEditingDashboardChart();
-            const unsavedDashboardTilesRaw = sessionStorage.getItem(
-                'unsavedDashboardTiles',
+        if (isDashboardLoading) return;
+        if (dashboardTiles) return;
+
+        setDashboardTiles(dashboard?.tiles ?? []);
+    }, [isDashboardLoading, dashboard, dashboardTiles, setDashboardTiles]);
+
+    useEffect(() => {
```

Comment:
This `useEffect` was quite complex; it was firing unpredictably and often doubled. I've made it a bit nicer, but we should refactor the entire "create chart from dashboard" flow at some point to make it less error-prone.

Reply:
Definitely.